src/strat_engine/blockdev.rs

```diff
@@ -320,8 +322,8 @@ impl BlockDev {
         buf[4..20].clone_from_slice(STRAT_MAGIC);
         LittleEndian::write_u64(&mut buf[20..28], *self.sectors);
         // no flags yet
-        buf[32..64].clone_from_slice(self.pool_id.as_bytes());
-        buf[64..96].clone_from_slice(self.id.as_bytes());
+        buf[32..64].clone_from_slice(self.pool_uuid.as_bytes());
+        buf[64..96].clone_from_slice(self.dev_uuid.as_bytes());
```

Comment:
instead of
```
buf[32..64].clone_from_slice(self.pool_uuid.as_bytes());
```
we want
```
buf[32..64].clone_from_slice(self.pool_uuid.simple().to_string().as_bytes());
```
Fixing and rebasing... (nice catch!)